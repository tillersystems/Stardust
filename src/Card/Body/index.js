import styled from 'styled-components';

/**
 * Body
 *
 * This component is in charge of display
 * the body of Card component
 *
 * @return {jsx}
 */
const Body = styled.div`
  flex: 1 1 0;
  overflow-y: auto;
`;

export default Body;
