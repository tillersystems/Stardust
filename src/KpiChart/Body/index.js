import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * KpiChart Body
 *
 * This component is in charge of displaying
 * the body of a kpi chart
 *
 * @param {string} children // children.
 * @param {string} className // className.
 *
 * @return {jsx}
 */
const KpiChartBody = ({ children, className }) => <div className={className}>{children}</div>;

/**
 * PropTypes Validation
 */
const { node, string } = PropTypes;
KpiChartBody.propTypes = {
  children: node,
  className: string,
};

/**
 * Default props
 */
KpiChartBody.defaultProps = {
  children: null,
  className: '',
};

export default styled(KpiChartBody)`
  height: 37.3rem;
`;
