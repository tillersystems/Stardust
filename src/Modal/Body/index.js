import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ModalBody = ({ children, className }) => <div className={className}>{children}</div>;

/**
 * PropTypes Validation
 */
const { node, string } = PropTypes;
ModalBody.propTypes = {
  children: node,
  className: string,
};

/**
 * Default props
 */
ModalBody.defaultProps = {
  children: null,
  className: '',
};

export default styled(ModalBody)`
  width: 100%;
  height: auto;
  flex: 1;
  margin: ${({ center }) => (center ? 'auto' : 'initial')};
`;
