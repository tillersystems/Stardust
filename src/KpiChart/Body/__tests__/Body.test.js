import React from 'react';

import Body from '..';

describe('<KpiChart.Body />', () => {
  test('should render without a problem', () => {
    const { container } = render(<Body>Body</Body>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
