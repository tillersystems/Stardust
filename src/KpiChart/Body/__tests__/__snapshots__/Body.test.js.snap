// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KpiChart.Body /> should render without a problem 1`] = `
.c0 {
  height: 37.3rem;
}

<div
  class="c0"
>
  Body
</div>
`;
